R-proj/src/rounding.cpp

```diff
@@ -42,55 +53,64 @@ Rcpp::NumericMatrix rounding (Rcpp::NumericMatrix A, unsigned int walk_len, bool
             NN=false,
             birk=false,
             verbose=false,
-            coordinate=coord;
+            coordinate=true, ball_walk = false;
+    NT delta = -1.0;
 
-    unsigned int m = A.nrow() - 1;
-    unsigned int n = A.ncol() - 1;
+    unsigned int n = P.field("dimension");
     unsigned int rnum = std::pow(1.0,-2.0) * 400 * n * std::log(n);
-    std::vector <std::vector<NT> > Pin(m + 1, std::vector<NT>(n + 1));
+    unsigned int walkL = 10+n/10;
+    if(walk_length.isNotNull()) walkL = Rcpp::as<unsigned int>(walk_length);
 
-    for (unsigned int i = 0; i < m + 1; i++) {
-        for (unsigned int j = 0; j < n + 1; j++) {
-            Pin[i][j] = A(i, j);
-        }
+    std::pair <Point, NT> InnerBall;
+    Rcpp::NumericMatrix Mat;
+    int type = P.field("type");
+    if (type==1) {
+        // Hpolytope
+        HP.init(n, Rcpp::as<MT>(P.field("A")), Rcpp::as<VT>(P.field("b")));
+        InnerBall = HP.ComputeInnerBall();
+    } else if (type==2) {
+        VP.init(n, Rcpp::as<MT>(P.field("V")), VT::Ones(Rcpp::as<MT>(P.field("V")).rows()));
+        InnerBall = VP.ComputeInnerBall();
+    } else if (type == 3) {
+        // Zonotope
+        ZP.init(n, Rcpp::as<MT>(P.field("G")), VT::Ones(Rcpp::as<MT>(P.field("G")).rows()));
+        InnerBall = ZP.ComputeInnerBall();
+    } else {
+        throw Rcpp::exception("Wrong polytope input");
     }
-    // construct polytope
-    if (Zono) {
-        ZP.init(Pin);
-    } else if (!Vpoly) {
-        HP.init(Pin);
+
+    if(!WalkType.isNotNull() || Rcpp::as<std::string>(WalkType).compare(std::string("CDHR"))==0){
+        coordinate = true;
```

Comment:
why not `enum Walk { random, coordinate, ball }` instead of using many `bool` vars? Then `WalkType` could also be an enum an avoid string comparisons

Reply:
Rcpp does not support enum for WalkType.